### PR TITLE
Frontend split receive component into bitbox02 and legacy (bitbox01)

### PR DESCRIPTION
**Pull request by thisconnect**
This has some duplicated code, but it will allow to completely revamp the receive view for BitBox02 and keep the legacy BitBox01 receive flow as it is.

### CODE REVIEW

File: frontends/web/src/routes/account/receive/receive.tsx [hunk 1]
```diff
@@ -284,20 +275,10 @@ export const Receive: FunctionComponent<Props> = ({
...
-          </>
-        )}
-      </Guide>
+      <ReceiveGuide
+        hasMultipleAddresses={currentAddresses && currentAddresses.length > 1}
+        hasDifferentFormats={receiveAddresses && receiveAddresses.length > 1}
+      />
     </div>
   );
 };
```
Comment (thisconnect) [inline frontends/web/src/routes/account/receive/receive.tsx:284]: maybe there could be other parts removed that are only relelvant to BB01 but I am not sure, cc @benma

File: frontends/web/src/routes/account/receive/receive-legacy.tsx [hunk 2]
```diff
@@ -0,0 +1,303 @@
+/**
```
Comment (benma) [inline frontends/web/src/routes/account/receive/receive-legacy.tsx:1]: i'd name the file `receive-bb01.tsx` for clarity.

File: frontends/web/src/routes/account/receive/receive-legacy.tsx [hunk 3]
```diff
@@ -64,7 +62,6 @@ const getIndexOfMatchingScriptType = (
 export const Receive: FunctionComponent<Props> = ({
   accounts,
   code,
-  devices,
```
Comment (benma) [inline frontends/web/src/routes/account/receive/receive-legacy.tsx]: the devices are still passed from receive/index.ts, right? typescript doesn't mind?

File: frontends/web/src/routes/account/receive/receive.tsx [hunk 4]
```diff
@@ -76,9 +70,8 @@ export const Receive: FunctionComponent<Props> = ({
   const [currentAddresses, setCurrentAddresses] = useState<accountApi.IReceiveAddress[]>();
   const [currentAddressIndex, setCurrentAddressIndex] = useState<number>(0);
 
-  const device = deviceIDs.length ? devices[deviceIDs[0]] : undefined;
   const account = accounts.find(({ code: accountCode }) => accountCode === code);
-  const verifyLabel = useVerfiyLabel(device);
+  const verifyLabel = useVerfiyLabel('bitbox02');
```
Comment (benma) [inline frontends/web/src/routes/account/receive/receive.tsx:73]: could remove that function and inline `t('receive.verifyBitBox02')`

**Review by benma — APPROVED** [review]
utACK

File: frontends/web/src/routes/account/receive/receive-legacy.tsx (hunk 3 above)
Comment (thisconnect) [reply frontends/web/src/routes/account/receive/receive-legacy.tsx]: I believe the short answer is it matches so TypeScript is happy.

https://stackoverflow.com/questions/49580725/is-it-possible-to-restrict-typescript-object-to-contain-only-properties-defined

https://github.com/Microsoft/TypeScript/issues/12936

Comment (thisconnect) [reply frontends/web/src/routes/account/receive/receive-legacy.tsx]: There is an eslint rule that would help, "react/no-unused-prop-types": "error" and catches a few cases, but unfortunately it has a few cases with the load decorator where it fails to recognise that those props are actually used and for some reason it does not catch this case :/

 I'll open a separate PR

Comment (thisconnect) [reply frontends/web/src/routes/account/receive/receive-legacy.tsx]: https://github.com/digitalbitbox/bitbox-wallet-app/pull/1733

**Comment by Beerosagos** [comment]
tACK! :+1:

**Pull request merged**